python/lib/communication/dmod/communication/data_transmit_message.py

```diff
@@ -18,64 +29,14 @@ class DataTransmitMessage(AbstractInitRequest):
     ::class:`str` object.  However, instances can be initialized using either ::class:`str` or ::class:`bytes` data.
     """
 
-    _KEY_SERIES_UUID = 'series_uuid'
+    event_type: ClassVar[MessageEventType] = MessageEventType.DATA_TRANSMISSION
 
-    event_type: MessageEventType = MessageEventType.DATA_TRANSMISSION
+    data: str = Field(description="The data carried by this message, in decoded string form.")
+    is_last: bool = Field(False, description="Whether this is the last data transmission message in this series.")
 
-    @classmethod
-    def factory_init_from_deserialized_json(cls, json_obj: dict) -> Optional['DataTransmitMessage']:
-        try:
-            return cls(data=json_obj['data'], series_uuid=UUID(json_obj[cls._KEY_SERIES_UUID]),
-                       is_last=json_obj['is_last'])
-        except Exception as e:
-            return None
 
-    def __init__(self, data: Union[str, bytes], series_uuid: UUID, is_last: bool = False, *args, **kwargs):
-        super(DataTransmitMessage, self).__init__(*args, **kwargs)
-        self._data: str = data if isinstance(data, str) else data.decode()
-        self._series_uuid = series_uuid
-        self._is_last: bool = is_last
-
-    @property
-    def data(self) -> str:
-        """
-        The data carried by this message, in decoded string form.
-
-        Returns
-        -------
-        str
-            The data carried by this message, in decoded string form.
-        """
-        return self._data
-
-    @property
-    def is_last(self) -> bool:
-        """
-        Whether this is the last data transmission message in this series.
-
-        Returns
-        -------
-        bool
-            Whether this is the last data transmission message in this series.
-        """
-        return self._is_last
-
-    @property
-    def series_uuid(self) -> UUID:
-        """
-        A unique id for the collective series of transmission message this instance is a part of.
-
-        The expectation is that a larger amount of data will be broken up into multiple messages in a series.
-
-        Returns
-        -------
-        UUID
-            A unique id for the collective series of transmission message this instance is a part of.
-        """
-        return self._series_uuid
-
-    def to_dict(self) -> Dict[str, Union[str, Number, dict, list]]:
-        return {'data': self.data, self._KEY_SERIES_UUID: str(self.series_uuid), 'is_last': self.is_last}
+class DataTransmitResponseBody(DataTransmitUUID):
```

Comment:
I'm honestly not sure exactly what this is supposed to do here, beyond have the `series_uuid` property. In theory, we had `Response.data` for serializing type-specific data beyond the standard `success`, `reason`, and `message` _Response_ properties.  I can't find anything using it for more than `series_uuid`, which makes the dedicated wrapper type seem excessive (or I am missing something).

We really need to overhaul the messages and logic for moving the data around anyway; I've opened #338 as an issue for that.  I'm curious if in the mean time it would just make more sense to drop this type; @aaraney, any thoughts?